scripts/jest/matchers/toWarnDev.ts

```diff
@@ -0,0 +1,65 @@
+/* tslint:disable:no-console */
+import jestDiff from 'jest-diff';
+
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toWarnDev(expectedMessage?: string): R;
+    }
+  }
+}
+
+const matcher = {
+  toWarnDev: (callback: () => void, expectedMessage: string) => {
+    if (expectedMessage !== undefined && typeof expectedMessage !== 'string') {
+      throw new Error(
+        `toWarnDev() requires a parameter of type string but was given ${typeof expectedMessage}.`
+      );
+    }
+
+    // @ts-ignore
```

Comment:
is there a particular rule that can be ignored here?

Reply:
I created #3577 for this. TSLint isn't aware of the `globals` set for ESLint.